contracts/Removal.sol

```diff
@@ -613,17 +562,56 @@ contract Removal is
     return ids;
   }
 
+  /**
+   * @dev Called by `_createRemovals`, validates the new IDs to prevent minting a pre-existing ID,
+   * stores the project id in a mapping.
+   *
+   * @param id The removal ID being minted.
+   * @param projectId The project id for this removal.
+   */
   function _createRemoval(uint256 id, uint256 projectId) internal {
     _validateRemoval({id: id});
     _removalIdToProjectId[id] = projectId;
   }
 
+  /**
+   * @dev Burns `amount` of token ID `id` from the supplier address encoded in the ID.
+   *
+   * Emits a {RemovalReleased} event.
+   *
+   * @param id The token ID to burn.
+   * @param amount The amount to burn.
+   */
+  function _releaseFromSupplier(uint256 id, uint256 amount) internal {
```

Comment:
Unrelated to this PR but I wonder if calling these internal helpers `_burnFrom...` rather than `_releaseFrom...` would be clearer.

Reply:
thought about that too, but for consistency with the `Certificate` situation (which involves more than just burning) I'm going to leave all of these consistent.  They also emit events called `RemovalReleased`